src/lib/logger/impl.ml

```diff
@@ -282,27 +287,36 @@ module Consumer_registry = struct
     Consumer_tbl.add_multi t ~key:id ~data:{ processor; transport }
 
   let broadcast_log_message ~id msg =
-    Hashtbl.find_and_call t id
-      ~if_found:(fun consumers ->
-        List.iter consumers
-          ~f:(fun
-               { processor = Processor.T ((module Processor), processor)
-               ; transport = Transport.T ((module Transport), transport)
-               }
-             ->
-            match Processor.process processor msg with
-            | Some str ->
-                Transport.transport transport str
-            | None ->
-                ()))
-      ~if_not_found:(fun _ ->
-        let (Processor.T ((module Processor), processor)) = Processor.raw () in
-        let (Transport.T ((module Transport), transport)) =
-          Transport.stdout ()
+    let consumers =
+      match Hashtbl.find t id with
+      | Some consumers ->
+          consumers
+      | None ->
+          [ Lazy.force default_consumer ]
+    in
+    List.iter consumers ~f:(fun consumer ->
+        let { processor = Processor.T ((module Processor), processor)
+            ; transport = Transport.T ((module Transport), transport)
+            } =
+          consumer
         in
         match Processor.process processor msg with
         | Some str ->
-            Transport.transport transport str
+            if String.length str < max_log_line_length then
+              Transport.transport transport str
+            else
+              let max_log_line_error =
+                { msg with
+                  message =
+                    "<log message elided as it was exceeded the max log line \
```

Comment:
I'm worried that this will trigger when we log parties transactions with proofs, and we'll lose the ability to track them through the system. Can we split the log entry instead?